### DISCUSSION TITLE
Istanbul ignore not working with swc and jest

### DISCUSSION THREAD

**Discussion by flextremedev** [vercel/next.js#49504] [Help]
### Summary

I really would love to migrate my projects from babel to swc however currently there seems to be no way to make istanbul ignore comments work with the next jest configuration.

Steps to reproduce (already set up in the example):
1. Checkout with-jest example.
2. Configure coverage e. g. inside jest.config.js: collectCoverageFrom: ['**/**/*.{ts,tsx}'],
3. Run test command with --coverage flag.
4. Observe _app.tsx is uncovered.
5. Add /* istanbul ignore file */ to the top of _app.tsx. Alternatively add /* istanbul ignore next */ above the component declaration.
6. Observe _app.tsx is still uncovered.

Using coverageIgnorePatterns is not a viable option because oftentimes you only want to ignore specific statements.
Any help is much appreciated.

### Example

https://github.com/flextremedev/next.js/tree/coverage-example

**Comment by flextremedev** [comment]
After further investigation I've noticed this seems to be a problem with export statements. Variables which are not imported inside test runs are shown as uncovered even though /* istanbul ignore next */ is added above the export statement.

**Reply by ronebrandaops** [reply]
@flextremedev I have a similar scenario here. In my case, the module that imports the `export default` is not considered by jest during the test execution, hence it considers it as not used. 

Did you do any workaround for this?

**Reply by flextremedev** [reply]
Hey, I see three possible solutions or workarounds for this problem:
- Add istanbul ignore file to file your default export is located in (which is what I did to resolve most of my problems)
- If your file has more exports maybe extract the other exports so you don't ignore them when using ignore file
- Test your default export in isolation

All in all I've found this behaviour on the coverage quite helpful since it showed me exports which I wasn't using at all. I just removed these export statements and that stopped causing the problem as well.
